cpp/src/arrow/CMakeLists.txt

```diff
@@ -362,6 +363,7 @@ add_arrow_test(pretty_print-test)
 add_arrow_test(public-api-test)
 add_arrow_test(scalar-test)
 add_arrow_test(status-test)
+add_arrow_test(error_or_test)
```

Comment:
nit: `error_or-test`

Reply:
I thought we were converging on this but you are right it is currently inconsistent, I'll fix once we finalize a name.